src/tools/r2rdump/R2RSignature.cs

```diff
@@ -460,7 +468,23 @@ public string ReadTypeSignature()
         /// <param name="builder"></param>
         private void ParseSignature(StringBuilder builder)
         {
-            uint fixupType = ReadUInt();
+            uint fixupType = PeekByte();
+            bool moduleOverride = (fixupType & (uint)CORCOMPILE_FIXUP_BLOB_KIND.ENCODE_MODULE_OVERRIDE) != 0;
+            // Check first byte for a module override being encoded
+            if (moduleOverride)
+            {
+                builder.Append("ENCODE_MODULE_OVERRIDE @ ");
+                fixupType &= ~(uint)CORCOMPILE_FIXUP_BLOB_KIND.ENCODE_MODULE_OVERRIDE;
+                ReadByte();
+                uint moduleIndex = ReadByte();
+                builder.Append(string.Format(" Index:  {0:X2}", moduleIndex));
+            }
+            // If the most significant bit has not been set, 
+            else
+            {
+                fixupType = ReadUInt();
```

Comment:
This is weird; this is probably my pre-existing bug. Double-checking the CoreCLR code I believe the fixup type should be always read using ReadByte even though we don't observe the difference in practice because all fixup kinds have enum value less than 128; actually, if I calculate correctly, the last ENCODE_METHOD_NATIVE_ENTRY in corcompile.h should have index 0x5C. This should also simplify the code as you will no longer need to peek at the first byte, you can just read it into fixupType right away (and possibly just subsequently clear the 7th bit if necessary).

Reply:
Please go over the relevant CoreCLR parsing code for yourself, don't just blindly follow my comment especially as you can clearly see I messed this up myself in the first place. I need to double-check whether I haven't messed this up in the CPAOT compiler too :-).